### PR TITLE
showed msg regarding max supply amount reached

### PR DESCRIPTION
**Pull request by smartdevcom**
image

### CODE REVIEW

**Comment by peetzweg** [comment]
Looks clean and easy, nice! 👍 
I guess we could also just hide the other tools `StatsColumn` and `Colleral toggle`.

File: packages/ui/utils/fetchMaxAmount.ts [hunk 1]
```diff
@@ -48,7 +48,16 @@ export const fetchMaxAmount = async (
...
 ): Promise<BigNumber> => {
   if (mode === FundOperationMode.SUPPLY) {
-    return await fetchTokenBalance(asset.underlyingToken, midasSdk, address);
+    const tokenBalance = await fetchTokenBalance(asset.underlyingToken, midasSdk, address);
+
+    if (comptrollerAddress) {
+      const comptroller = midasSdk.createComptroller(comptrollerAddress);
+      const supplyCap = await comptroller.callStatic.supplyCaps(asset.cToken);
+
+      return supplyCap.gt(constants.Zero) && supplyCap.lte(tokenBalance) ? supplyCap : tokenBalance;
```
[the remark this answers is not in the document]
Comment (peetzweg) [reply packages/ui/utils/fetchMaxAmount.ts:57]: Math.min probalby won't work with BigNumbers, but used here to demonstrate the logic :D

File: packages/ui/utils/fetchMaxAmount.ts [hunk 2]
```diff
@@ -48,7 +48,16 @@ export const fetchMaxAmount = async (
   comptrollerAddress?: string
 ): Promise<BigNumber> => {
   if (mode === FundOperationMode.SUPPLY) {
```
Comment (peetzweg) [inline packages/ui/utils/fetchMaxAmount.ts:50]: We also need this feature for `FundOperationMode.BORROW` for `BorrowCaps`

File: packages/ui/utils/fetchMaxAmount.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (smartdevcom) [reply packages/ui/utils/fetchMaxAmount.ts:57]: Fixed

**Comment by smartdevcom** [comment]
> Looks clean and easy, nice! +1 I guess we could also just hide the other tools `StatsColumn` and `Colleral toggle`.

image

File: packages/ui/utils/fetchMaxAmount.ts (hunk 2 above)
Comment (smartdevcom) [reply packages/ui/utils/fetchMaxAmount.ts:50]: I will do it in separated PR

**Pull request merged**